EssentialFeed/EssentialFeedTests/Feed API/LoadFeedImageCommentsFromRemoteUseCaseTests.swift

```diff
@@ -0,0 +1,225 @@
+		
+		XCTAssertEqual(client.requestedURLs, [url, url])
+	}
+	
+	func test_loadImageCommentsFromURL_deliversConnectivityErrorOnClientError() {
+		let (sut, client) = makeSUT()
+		let clientError = NSError(domain: "a client error", code: 0)
+		
+		expect(sut, toCompleteWith: failure(.connectivity), when: {
+			client.complete(with: clientError)
+		})
+	}
+	
+	func test_loadImageCommentsFromURL_deliversInvalidDataErrorOnNon2xxHTTPResponse() {
+		let (sut, client) = makeSUT()
+		
+		let samples = [199, 150, 300, 400, 500]
+		
+		samples.enumerated().forEach { index, code in
+			expect(sut, toCompleteWith: failure(.invalidData), when: {
+				client.complete(withStatusCode: code, data: makeItemsJSON([]), at: index)
+			})
+		}
+	}
+	
+	func test_loadImageCommentsFromURL_deliversInvalidDataErrorOn2xxHTTPResponseWithEmptyData() {
+		let (sut, client) = makeSUT()
+		
+		let samples = [200, 201, 299]
+		
+		samples.enumerated().forEach { index, code in
+			expect(sut, toCompleteWith: failure(.invalidData), when: {
+				let emptyData = Data()
+				client.complete(withStatusCode: code, data: emptyData, at: index)
+			})
+		}
+	}
+	
+	func test_load_deliversErrorOn2xxHTTPResponseWithInvalidJSON() {
+		let (sut, client) = makeSUT()
+		
+		let samples = [200, 201, 299]
+		
+		samples.enumerated().forEach { index, code in
+			expect(sut, toCompleteWith: failure(.invalidData), when: {
+				let invalidJSON = Data("invalid json".utf8)
+				client.complete(withStatusCode: code, data: invalidJSON, at: index)
+			})
+		}
+	}
+	
+	func test_load_deliversNoItemsOn2xxHTTPResponseWithEmptyJSONList() {
+		let (sut, client) = makeSUT()
+		
+		let samples = [200, 201, 299]
+		
+		samples.enumerated().forEach { index, code in
+			expect(sut, toCompleteWith: .success([]), when: {
+				let emptyListJSON = makeItemsJSON([])
+				client.complete(withStatusCode: code, data: emptyListJSON, at: index)
+			})
+		}
+	}
+	
+	func test_load_deliversItemsOn2xxHTTPResponseWithJSONItems() {
+		let (sut, client) = makeSUT()
+		
+		let item1 = makeItem(id: UUID(),
+							 message: "a message",
+							 createdAt: (Date(timeIntervalSince1970: 1598627222), "2020-08-28T15:07:02+00:00"),
+							 username: "a username")
+		
+		let item2 = makeItem(id: UUID(),
+							 message: "another message",
+							 createdAt: (Date(timeIntervalSince1970: 1577881882), "2020-01-01T12:31:22+00:00"),
+							 username: "another username")
+		
+		let items = [item1.model, item2.model]
+		
+		let samples = [200, 201, 299]
+		
+		samples.enumerated().forEach { index, code in
+			expect(sut, toCompleteWith: .success(items), when: {
+				let json = makeItemsJSON([item1.json, item2.json])
+				client.complete(withStatusCode: code, data: json, at: index)
+			})
+		}
+	}
+	
+	func test_cancelLoadImageCommentsURLTask_cancelsClientURLRequest() {
+		let image = uniqueImage()
+		let url = URL(string: "https://a-given-url.com/image/\(image.id.uuidString)/comments")!
+		let (sut, client) = makeSUT(image: image)
```

Comment:
The image and the image id are not needed in this test. You can pass *any* URL to the remote loader.

Remove the image and pass a URL instead:

```suggestion
		let url = URL(string: "https://a-given-url.com")!
		let (sut, client) = makeSUT(url: url)
```